consensus/scheme/rolldpos/endorsementmanager.go

```diff
@@ -375,6 +400,7 @@ func (m *endorsementManager) Cleanup(timestamp time.Time) error {
 	} else {
 		m.collections = map[string]*blockEndorsementCollection{}
 	}
+	m.cachedMintedBlk = nil
```

Comment:
every time, we "Prepare" a new round, the cachedMintedBlk will be clean up. So, this implementation may not work.

Reply:
Thanks for pointing it out! I fixed this issue by adding conditions when we clean up endorsementmanager. (Clean up only when the cached block timestamp is before round start time)